remoting/src/main/java/org/springframework/security/remoting/dns/JndiDnsResolver.java

```diff
@@ -61,16 +61,28 @@ public String resolveIpAddress(String hostname) {
 
 	@Override
 	public String resolveServiceEntry(String serviceType, String domain) {
-		return resolveServiceEntry(serviceType, domain, this.ctxFactory.getCtx());
+		return resolveServiceEntry(serviceType, domain, this.ctxFactory.getCtx()).getHostName();
 	}
 
 	@Override
 	public String resolveServiceIpAddress(String serviceType, String domain) {
 		DirContext ctx = this.ctxFactory.getCtx();
-		String hostname = resolveServiceEntry(serviceType, domain, ctx);
+		String hostname = resolveServiceEntry(serviceType, domain, ctx).getHostName();
 		return resolveIpAddress(hostname, ctx);
 	}
 
+	/**
+	 * @author Kathryn Newbould
+	 * @since 5.4.1
```

Comment:
This change will be in 5.5 , since 5.4.1 is a patch release and should only include bug fixes.
```suggestion
	 * @since 5.5
```